doc/reference/trd2-hil-design.md

```diff
@@ -256,20 +256,24 @@ impl Random for CachingRNG {
     self.busy.set(true);
     if self.cached_words.get() > 0 {
       // This tells the scheduler to issue a deferred procedure call,
-      // passing the "handle" the scheduler uses to keep track of it.
-      self.handle.map(|handle| self.deferred_call.set(*handle));
+      self.deferred_call.set();
     } else {
       self.request_more_randomness();
     }
   }
   ...
 }
 
-impl<'a> DynamicDeferredCallClient for CachingRNG<'a> {
-  fn call(&self, _handle: DeferredCallHandle) {
+impl<'a> DeferredCallClient for CachingRNG<'a> {
+  fn handle_deferred_call(&self) {
     let rbits = self.pop_cached_word();
     self.client.random_ready(rbits, Ok(()));
   }
+
+  // This function must be called during board initialization.
+  fn register(&'static self) {
+      self.deferred_call.register(self);
+  }
```

Comment:
ugh...feels so silly to have to make a whole new document to update the syntax of the code that it references

Reply:
Done, but this has made me think that TRD101 is flawed. Either code snippets that use internal Tock APIs should not be included in finalized TRDs or they should be allowed to change, since internal Tock APIs are not stable. The current process for things becoming obsolete has several flaws:

- both documents still live in the trds folder, and it is very easy for people to accidentally reference the old one without realizing it is obsolete (this has already happened with the Time TRD!)
- Links to documentation in PRs and issue discussions will commonly link to the obsoleted PR (this would not happen if we updated the TRDs in place)
- This process creates an incentive not to update code snippets in TRDs at all, since doing so will make even small API changes seem like bigger changes by creating huge diffs from having to copy over an entire TRD to make a small syntax change

Reply:
Yes, I was about to make this same comment but don't feel like this is an appropriate place to discuss it. Perhaps bring it up during the next core WG meeting?

Reply:
Yeah, I will.